### PR TITLE
Feature/custom header support

### LINKED ISSUE #50: Add ability to pass a custom header to AWS OAuth Proxy
We need the ability to pass in a custom header so we can call JD API with that same custom header.

### PR DESCRIPTION
**Pull request by nbodesa**
This adds the ability to pass in a querystring called allData=true|false. If true, this will add the No_Paging header onto the get requests to JD's API

resolves #50

### CODE REVIEW

File: local.params.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{
+  "OAuthFirstLegLambda": {
+    "API_URL": "https://partnerapi.deere.com/platform",
+    "AUTHORIZE_CALLBACK_URI": "saimyjd://oauth",
+    "CLIENT_KEY": "AQICAHgmK7RLzJGLsX027nGL/3YoG67Jh4Mxgt5hOhZTg/kH0AEBuo41ZST4tFLvnatiZAe6AAAAgDB+BgkqhkiG9w0BBwagcTBvAgEAMGoGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQMIOTJJObOrbPgZE9mAgEQgD2caSEzBrt/W+/SyOTA7owW6nhDuHIohZreY9K0YhoNwmZuYZwcx2ev+bt1SaNdTkJf0T3frcP6ypUXiGn2",
+    "CLIENT_SECRET": "AQICAHgmK7RLzJGLsX027nGL/3YoG67Jh4Mxgt5hOhZTg/kH0AFXkaxExrS4VONgHtAjx8pnAAAAojCBnwYJKoZIhvcNAQcGoIGRMIGOAgEAMIGIBgkqhkiG9w0BBwEwHgYJYIZIAWUDBAEuMBEEDHIXFWKlwjyLUs/iCwIBEIBbKrDUS/E/+/W5FGXaeuhNf8z4qnErSZTVmnfOwhN8F0OdWZPYN/bLz721HuxPnl2XdDVkcktqsQQiuPGS4YOgnN6XTnMI2BV4zO/bxxmg/L2EFH+gHEL37+LHRA==",
```
Comment (mattcvincent) [inline local.params.json:6]: @nbodesa @teolandon Not sure if you were aware, but this is an open source project, so anything you commit, will be instantly searchable/viewable by the world.  There are bots that search for commits of AWS keys and SECRETS accidentally committed by users and immediately take action on those (usually bitcoin mining before you figure it out a few thousand dollars later).  I don't know the sensitivity of these secrets, but unless I misunderstand the intent, I suggest removing them from the commit and rotating the keys.

Comment (nbodesa) [reply local.params.json:6]: These keys are encrypted with KMS and the same ones found here: https://github.com/sourceallies/aws-oauth-1.0-proxy/tree/master/deploy/params.

File: test/App.test.js [hunk 2]
```diff
@@ -70,433 +77,522 @@ describe('Lambda handlers', () => {
...
-        .toHaveBeenCalledWith({...fakeEvent, ...expectedResponse});
+      expect(publishToSNSSuccess).toHaveBeenCalledWith({
+        ...fakeEvent,
+        ...expectedResponse,
+      });
     });
 
-    it('Should Send a unSuccessful Response To SNS Channel', async () => {
-      const OAuth = require('oauth');
+    it("Should Send a unSuccessful Response To SNS Channel", async () => {
```
Comment (blarmon) [inline test/App.test.js:394]:
```suggestion
    it("Should Send an Unsuccessful Response To SNS Channel", async () => {
```

Comment (nbodesa) [reply test/App.test.js]: Fixed

**Pull request merged**